File: src/ZKWeb.MVVMPlugins/MVVM.Angular.Website/static/src/modules/generated_module/services/website-manage-service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { AppApiService } from '@global_module/services/app-api-service';
import { rxResultConverter, rxErrorConverter } from '@core/utils/type-utils';
import { ActionResponseDto } from '@generated_module/dtos/action-response-dto';
import { WebsiteInfoOutputDto } from '@generated_module/dtos/website-info-output-dto';
import { WebsiteSettingsDto } from '@generated_module/dtos/website-settings-dto';
import { GridSearchResponseDto } from '@generated_module/dtos/grid-search-response-dto';
import { GridSearchRequestDto } from '@generated_module/dtos/grid-search-request-dto';

@Injectable()
/** 网站管理服务 */
export class WebsiteManageService {
    constructor(private appApiService: AppApiService) { }

    /** 清理缓存 */
    ClearCache(resultConverter?: rxResultConverter, errorConverter?: rxErrorConverter): Observable<ActionResponseDto> {
        return this.appApiService.call<ActionResponseDto>(
            "/api/WebsiteManageService/ClearCache",
            {
                method: "POST"
            }, resultConverter, errorConverter);
    }

    /** 获取网站信息 */
    GetWebsiteInfo(resultConverter?: rxResultConverter, errorConverter?: rxErrorConverter): Observable<WebsiteInfoOutputDto> {
        return this.appApiService.call<WebsiteInfoOutputDto>(
            "/api/WebsiteManageService/GetWebsiteInfo",
            {
                method: "POST"
            }, resultConverter, errorConverter);
    }

    /** 获取网站信息 */
    GetWebsiteSettings(resultConverter?: rxResultConverter, errorConverter?: rxErrorConverter): Observable<WebsiteSettingsDto> {
        return this.appApiService.call<WebsiteSettingsDto>(
            "/api/WebsiteManageService/GetWebsiteSettings",
            {
                method: "POST"
            }, resultConverter, errorConverter);
    }

    /** 保存网站信息 */
    SaveWebsiteSettings(dto: WebsiteSettingsDto, resultConverter?: rxResultConverter, errorConverter?: rxErrorConverter): Observable<ActionResponseDto> {
        return this.appApiService.call<ActionResponseDto>(
            "/api/WebsiteManageService/SaveWebsiteSettings",
            {
                method: "POST",
                body: { dto }
            }, resultConverter, errorConverter);
    }

    /** 搜索定时任务 */
    SearchScheduledTasks(request: GridSearchRequestDto, resultConverter?: rxResultConverter, errorConverter?: rxErrorConverter): Observable<GridSearchResponseDto> {
        return this.appApiService.call<GridSearchResponseDto>(
            "/api/WebsiteManageService/SearchScheduledTasks",
            {
                method: "POST",
                body: { request }
            }, resultConverter, errorConverter);
    }

    /** 搜索定时任务记录 */
    SearchScheduledTaskLogs(request: GridSearchRequestDto, resultConverter?: rxResultConverter, errorConverter?: rxErrorConverter): Observable<GridSearchResponseDto> {
        return this.appApiService.call<GridSearchResponseDto>(
            "/api/WebsiteManageService/SearchScheduledTaskLogs",
            {
                method: "POST",
                body: { request }
            }, resultConverter, errorConverter);
    }
}

File: src/BusinessPlugins/OrganizationModule/src/Application/Services/WebsiteManageService.cs
﻿using AutoMapper;
using BusinessPlugins.OrganizationModule.Application.Dtos;
using BusinessPlugins.OrganizationModule.Components.ActionFilters;
using BusinessPlugins.OrganizationModule.Components.GenericConfigs;
using BusinessPlugins.OrganizationModule.Domain.Entities.Interfaces;
using BusinessPlugins.OrganizationModule.Domain.Services;
using InfrastructurePlugins.BaseModule.Application.Dtos;
using InfrastructurePlugins.BaseModule.Application.Extensions;
using InfrastructurePlugins.BaseModule.Application.Services.Bases;
using InfrastructurePlugins.TaskSchedulerModule.Application.Dtos;
using InfrastructurePlugins.TaskSchedulerModule.Domain.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using ZKWeb.Cache;
using ZKWeb.Localize;
using ZKWeb.Plugin;
using ZKWebStandard.Ioc;
using ZKWebStandard.Utils;

namespace BusinessPlugins.OrganizationModule.Application.Services
{
    /// <summary>
    /// 网站管理服务
    /// </summary>
    [ExportMany, SingletonReuse, Description("网站管理服务")]
    public class WebsiteManageService : ApplicationServiceBase
    {
        private IList<ICacheCleaner> _cacheCleaners;
        private PluginManager _pluginManager;
        private GenericConfigManager _configManager;

        public WebsiteManageService(
            IEnumerable<ICacheCleaner> cacheCleaners,
            PluginManager pluginManager,
            GenericConfigManager configManager)
        {
            _cacheCleaners = cacheCleaners.ToList();
            _pluginManager = pluginManager;
            _configManager = configManager;
        }

        [Description("清理缓存")]
        [CheckPrivilege(typeof(ICanUseAdminPanel))]
        public ActionResponseDto ClearCache()
        {
            foreach (var cleaner in _cacheCleaners)
            {
                cleaner.ClearCache();
            }
            GC.Collect();
            return ActionResponseDto.CreateSuccess("Clear Cache Successfully");
        }

        [Description("获取网站信息")]
        [CheckPrivilege(typeof(ICanUseAdminPanel))]
        public WebsiteInfoOutputDto GetWebsiteInfo()
        {
            var websiteInfo = new WebsiteInfoOutputDto();
            websiteInfo.ZKWebVersion = ZKWeb.Application.Version.ToString();
            websiteInfo.ZKWebFullVersion = ZKWeb.Application.FullVersion.ToString();
            websiteInfo.MemoryUsage = FileUtils.GetSizeDisplayName(SystemUtils.GetUsedMemoryBytes());
            foreach (var plugin in _pluginManager.Plugins)
            {
                websiteInfo.Plugins.Add(new PluginInfoOutputDto()
                {
                    DirectoryName = plugin.DirectoryName(),
                    Name = new T(plugin.Name),
                    Version = plugin.VersionObject().ToString(),
                    FullVersion = plugin.Version,
                    Description = new T(plugin.Description)
                });
            }
            return websiteInfo;
        }

        [Description("获取网站信息")]
        [CheckPrivilege(typeof(ICanUseAdminPanel))]
        public WebsiteSettingsDto GetWebsiteSettings()
        {
            var settings = _configManager.GetData<WebsiteSettings>();
            var dto = Mapper.Map<WebsiteSettingsDto>(settings);
            return dto;
        }

        [Description("保存网站信息")]
        [CheckPrivilege(typeof(IAmAdmin), "Settings:WebsiteSettings")]
        public ActionResponseDto SaveWebsiteSettings(WebsiteSettingsDto dto)
        {
            var settings = Mapper.Map<WebsiteSettings>(dto);
            _configManager.PutData(settings);
            return ActionResponseDto.CreateSuccess("Saved Successfully");
        }

        [Description("搜索定时任务")]
        [CheckPrivilege(typeof(IAmAdmin), "ScheduledTask:View")]
        public GridSearchResponseDto SearchScheduledTasks(GridSearchRequestDto request)
        {
            return request.BuildResponse<ScheduledTask, ScheduledTaskOutputDto, string>()
                .FilterKeywordWith(t => t.Id)
                .ToResponse();
        }

        [Description("搜索定时任务记录")]
        [CheckPrivilege(typeof(IAmAdmin), "ScheduledTask:View")]
        public GridSearchResponseDto SearchScheduledTaskLogs(GridSearchRequestDto request)
        {
            return request.BuildResponse<ScheduledTaskLog, ScheduledTaskLogOutputDto, Guid>()
                .FilterKeywordWith(t => t.Task.Id)
                .FilterKeywordWith(t => t.Error)
                .FilterColumnWith(
                    nameof(ScheduledTaskLogOutputDto.TaskId),
                    (c, q) => q.Where(t => t.Task.Id.Contains((string)c.Value)))
                .ToResponse();
        }

        [Description("获取当前的Api函数信息")]
        public string GetApiMethodInfo()
        {
            var info = Context.GetApiMethodInfo();
            return info.Name;
        }
    }
}
